src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1467,11 +1469,18 @@ protected void ResetEngineInternal()
 
             EngineController.MessageLogged += LogMessage;
             EngineController.TraceReconcliationComplete += EngineController_TraceReconcliationComplete;
+            EngineController.RequestCustomNodeRegistration += EngineController_RequestCustomNodeRegistration;
```

Comment:
I wonder if we should just move all these subscriptions to the constructor for `EngineController`?